File: invoicepdf/backend/check_task_logs.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
检查任务日志 - 查看任务执行详情
"""

import sys
import os
from datetime import datetime

if sys.platform == 'win32':
    import io
    sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')
    sys.stderr = io.TextIOWrapper(sys.stderr.buffer, encoding='utf-8')

try:
    import psycopg2
    from psycopg2.extras import RealDictCursor
except ImportError:
    print("[错误] 需要安装 psycopg2: pip install psycopg2-binary")
    sys.exit(1)

DB_HOST = "219.151.188.129"
DB_PORT = "50510"
DB_USER = "postgres"
DB_PASSWORD = "Post.&0055"
DB_NAME = "app"

def check_task_logs(task_id="0fd519c9-7716-42b2-8091-dae1cb3f535b"):
    """检查特定任务的详细信息"""
    
    print("=" * 80)
    print(f"检查任务详情: {task_id}")
    print("=" * 80)
    
    conn = psycopg2.connect(
        host=DB_HOST,
        port=DB_PORT,
        user=DB_USER,
        password=DB_PASSWORD,
        database=DB_NAME
    )
    cur = conn.cursor(cursor_factory=RealDictCursor)
    
    try:
        # 查询任务详情
        cur.execute("""
            SELECT 
                t.*,
                i.invoice_no,
                f.file_name,
                f.external_file_id,
                f.file_path,
                llm.name as config_name,
                llm.endpoint,
                llm.api_key,
                llm.app_type,
                llm.workflow_id
            FROM recognition_task t
            JOIN invoice i ON t.invoice_id = i.id
            JOIN invoice_file f ON i.file_id = f.id
            LEFT JOIN llm_config llm ON (t.params->>'model_config_id')::uuid = llm.id
            WHERE t.id = %s;
        """, (task_id,))
        
        task = cur.fetchone()
        
        if not task:
            print(f"[错误] 未找到任务: {task_id}")
            return
        
        print(f"\n任务基本信息:")
        print(f"  任务ID: {task['id']}")
        print(f"  任务编号: {task['task_no']}")
        print(f"  状态: {task['status']}")
        print(f"  创建时间: {task['create_time']}")
        print(f"  开始时间: {task['start_time']}")
        print(f"  结束时间: {task['end_time']}")
        if task['start_time']:
            elapsed = datetime.now() - task['start_time']
            print(f"  已运行时长: {elapsed}")
        print(f"  错误代码: {task['error_code']}")
        print(f"  错误消息: {task['error_message']}")
        
        print(f"\n文件信息:")
        print(f"  文件名: {task['file_name']}")
        print(f"  文件路径: {task['file_path']}")
        print(f"  external_file_id: {task['external_file_id']}")
        
        print(f"\n模型配置:")
        print(f"  配置名称: {task['config_name']}")
        print(f"  API端点: {task['endpoint']}")
        print(f"  API Key: {'*' * 20 if task['api_key'] else '未设置'}")
        print(f"  应用类型: {task['app_type']}")
        print(f"  工作流ID: {task['workflow_id']}")
        
        print(f"\n任务参数:")
        import json
        params = task['params']
        if params:
            print(json.dumps(params, ensure_ascii=False, indent=2))
        
        # 检查是否有识别结果
        cur.execute("""
            SELECT id, status, create_time
            FROM recognition_result
            WHERE task_id = %s;
        """, (task_id,))
        
        result = cur.fetchone()
        if result:
            print(f"\n识别结果:")
            print(f"  结果ID: {result['id']}")
            print(f"  状态: {result['status']}")
            print(f"  创建时间: {result['create_time']}")
        else:
            print(f"\n识别结果: 无")
        
        # 检查文件是否存在
        import os
        if task['file_path']:
            file_exists = os.path.exists(task['file_path'])
            print(f"\n文件检查:")
            print(f"  文件路径存在: {file_exists}")
            if file_exists:
                file_size = os.path.getsize(task['file_path'])
                print(f"  文件大小: {file_size} 字节")
        
    except Exception as e:
        print(f"\n[错误] 检查失败: {str(e)}")
        import traceback
        traceback.print_exc()
    finally:
        cur.close()
        conn.close()

if __name__ == "__main__":
    task_id = sys.argv[1] if len(sys.argv) > 1 else "0fd519c9-7716-42b2-8091-dae1cb3f535b"
    check_task_logs(task_id)

File: invoicepdf/backend/check_task_logs.js
// 检查任务详细状态和可能的错误
// 在浏览器控制台运行

const taskId = "TASK-20260128110745-ee3a2a5b";
const token = localStorage.getItem('access_token');

(async function() {
    console.log('=== 检查任务状态 ===');
    console.log(`任务编号: ${taskId}\n`);
    
    const res = await fetch('http://localhost:8000/api/v1/invoices/recognition-tasks?limit=100', {
        headers: { 'Authorization': `Bearer ${token}` }
    });
    const data = await res.json();
    const task = data.data.find(t => t.task_no === taskId);
    
    if (!task) {
        console.log('未找到任务');
        return;
    }
    
    const startTime = task.start_time ? new Date(task.start_time) : null;
    const now = new Date();
    const elapsed = startTime ? Math.round((now - startTime) / 1000) : 0;
    const minutes = Math.floor(elapsed / 60);
    const seconds = elapsed % 60;
    
    console.log('任务信息:');
    console.log(`  状态: ${task.status}`);
    console.log(`  创建时间: ${task.create_time}`);
    console.log(`  开始时间: ${task.start_time || 'N/A'}`);
    console.log(`  已处理时间: ${minutes}分${seconds}秒`);
    
    console.log(`\n配置信息:`);
    console.log(`  模型: ${task.model_name}`);
    console.log(`  模板ID (任务字段): ${task.template_id || 'N/A'}`);
    console.log(`  模板ID (params中): ${task.params?.template_id || 'N/A'}`);
    console.log(`  提示词: ${task.params?.template_prompt ? '✓ 存在 (' + task.params.template_prompt.length + '字符)' : '✗ 不存在'}`);
    
    console.log(`\n分析:`);
    if (elapsed > 600) {
        console.log(`  ⚠ 任务已处理超过10分钟，可能存在问题`);
        console.log(`  建议:`);
        console.log(`    1. 检查后端日志，查看是否有错误或超时`);
        console.log(`    2. 检查 DIFY API 是否正常响应`);
        console.log(`    3. 如果任务卡住，可以考虑取消并重新创建`);
    } else if (elapsed > 300) {
        console.log(`  ⚠ 任务已处理超过5分钟，可能较慢但仍在处理中`);
    } else {
        console.log(`  ✓ 任务正在正常处理中`);
    }
    
    // 检查是否有错误信息
    if (task.params?.error_message) {
        console.log(`\n错误信息: ${task.params.error_message}`);
    }
    if (task.error_message) {
        console.log(`\n任务错误: ${task.error_message}`);
    }
    if (task.error_code) {
        console.log(`\n错误代码: ${task.error_code}`);
    }
    
    console.log(`\n建议操作:`);
    console.log(`  1. 查看后端控制台日志，查找任务相关的日志`);
    console.log(`  2. 如果看到 "从模板对象获取到模板提示词" 的日志，说明修复生效`);
    console.log(`  3. 如果任务继续卡住，可以等待或取消后重新创建`);
})();

